src/components/button/Etaps.tsx

```diff
@@ -1,84 +1,138 @@
 import { useState, useEffect } from "react";
-import { database } from "../../utils/firebase/firebase.config.jsx";
-import {
-  collection,
-  getDocs,
-  DocumentData,
-  query,
-  where,
-  onSnapshot,
-} from "firebase/firestore";
-import Activites from "../activities/Activities";
-
-interface IEtap {
+import { database } from "../../utils/firebase/firebase.config"; // import storage
+import { collection, getDocs } from "firebase/firestore";
+import { ref, getDownloadURL, getStorage } from "firebase/storage";
+import { getApp } from "firebase/app";
+import Activities from "../activities/Activities";
+
+interface Etap {
   id: string;
   name: string;
   sort: number;
+  icon: string;
 }
 
 function Etaps() {
-  const [etaps, setEtaps] = useState<IEtap[]>([]);
+  const [etaps, setEtaps] = useState<Etap[]>([]);
   const [etapId, setEtapId] = useState<string | null>(null);
-  const [isButtonDisabled, setIsButtonDisabled] = useState<boolean>(false);
+  const [activitiesByEtap, setActivitiesByEtap] = useState<
+    Record<string, { etap_id: string; id: string }[]>
+  >({});
+  const [userActivityIds, setUserActivityIds] = useState<string[]>([]);
 
   useEffect(() => {
-    const getEtaps = async (): Promise<void> => {
+    async function getEtaps() {
       const etapsRef = collection(database, "etaps");
       const etapsData = await getDocs(etapsRef);
 
-      const etapsArray: IEtap[] = [];
-      etapsData.forEach((doc: DocumentData) => {
-        const etap: IEtap = {
+      //make array with etpaid and activities
+      const activitiesRef = collection(database, "activities");
+      const activitiesData = await getDocs(activitiesRef);
+      const activitiesByEtap = activitiesData.docs.reduce<
+        Record<string, { etap_id: string; id: string }[]>
+      >((acc, doc) => {
+        const etapId = doc.data().etap_id;
+
+        if (!acc[etapId]) {
+          acc[etapId] = [];
+        }
+
+        acc[etapId].push({
+          etap_id: etapId,
           id: doc.id,
-          name: doc.data().name,
-          sort: doc.data().sort,
-        };
-        etapsArray.push(etap);
-      });
+        });
+
+        return acc;
+      }, {});
+
+      setActivitiesByEtap(activitiesByEtap);
+      //make array from user_activities with etpaid and activityid
+      const userActivitiesRef = collection(database, "user_activities");
+      const useractivitiesData = await getDocs(userActivitiesRef);
+      const userActivityIds = useractivitiesData.docs.map(
+        (doc) => doc.data().user_activity_id
+      );
+
+      setUserActivityIds(userActivityIds);
+
+      //icon part, take from storage and push in button
+      const etapsArray: Etap[] = await Promise.all(
+        etapsData.docs.map(async (doc) => {
+          const etap = {
+            id: doc.id,
+            name: doc.data().name,
+            sort: doc.data().sort,
+            icon: "",
+          };
```

Comment:
w etapsData, można chyba tylko po id:doc.id dobrać się do dokumentu a następnie gdzie masz potrzebę to wyciągnąć, element który potrzebujesz.

Reply:
j.w.